packages/editor/src/components/block-styles/index.js

```diff
@@ -146,6 +146,12 @@ export default compose( [
 		const { getBlock } = select( 'core/editor' );
 		const { getBlockStyles } = select( 'core/blocks' );
 		const block = getBlock( clientId );
+
```

Comment:
It seems like this check here should not be needed because when the block is removed <BlockStyles component should not be rendered at all, because of this condition on packages/editor/src/components/block-inspector/index.js
```
	/*
	 * If the selected block is of an unregistered type, avoid showing it as an actual selection
	 * because we want the user to focus on the unregistered block warning, not block settings.
	 */
	if ( ! blockType || ! selectedBlockClientId || isSelectedBlockUnregistered ) {
		return <span className="editor-block-inspector__no-blocks">{ __( 'No block selected.' ) }</span>;
	}
```
It may be related to a race condition related to async changes cc: @youknowriad.

Another alternative to an explicit condition that also fixes the problem is: 
```
	withSelect( ( select, { clientId } ) => {
		const { getBlockName, getBlockAttributes } = select( 'core/editor' );
		const { getBlockStyles } = select( 'core/blocks' );
		const blockName = getBlockName( clientId );
		const blockType = getBlockType( blockName );
		const blockAttributes = getBlockAttributes( clientId );
		return {
			name: blockName,
			attributes: blockAttributes,
			className: ( blockAttributes && blockAttributes.className ) || '',
			styles: getBlockStyles( blockName ),
			type: blockType,
		};
	} ),
```

Reply:
Is an explicit conditional bad in this instance?